pkg/util/mount/mount_linux.go

```diff
@@ -430,3 +430,19 @@ func (mounter *SafeFormatAndMount) getDiskFormat(disk string) (string, error) {
 	// and MD RAID are reported as FSTYPE and caught above).
 	return "unknown data, probably partitions", nil
 }
+
+// IsSamefsGroup is called only for requests to mount an already mounted
+// volume. It checks if fsGroup of new mount request is the same or not.
+// It returns false if it not the same. It also returns current Gid of a path
+// provided for dir variable.
+func IsSamefsGroup(dir string, fsGroup *int64) (bool, int, error) {
+	s := &syscall.Stat_t{}
+	if err := syscall.Stat(dir, s); err != nil {
+		return false, 0, err
+	}
+	gidNew := 0
+	if fsGroup != nil {
+		gidNew = int(*fsGroup)
```

Comment:
chat with @msau42, the default gid may be volatile depending how kubelet is started, we should not default to gid 0.

Since fsGroup is used to set directory ownership and nil fsGroup is already validated there, we only need to focus the case where fsGroup!=nil. If fsGroup == nil, return false and nil here.